### ISSUE TITLE
Strip shortcodes in related post excerpts

### ISSUE THREAD

**Issue by barrykooij** [barrykooij/related-posts-for-wp#68]
Say I am viewing a Post, and see the Related Posts at the bottom. I've chosen to show Featured Image, and those display fine. A couple of the Related Posts have an image at the top. In the raw code of those Posts there is a shortcode for the caption material. Unfortunately, this raw caption code shows in the Related Posts excerpt, which looks kind of funny. :)

https://wordpress.org/support/topic/image-near-top-of-related-post-caption-shortcode?replies=3

**Comment by davechu** [comment]
Hi Barry! 
I appreciate your checking into this. Curiously, I fixed one of the Posts that balked (not this one) when I edited the code and found an old-school "more" tag. Once I removed that, the Post was "pulled into" Related Posts on another Post just fine. 

Here's some raw code of a Post that did not have "more", yet is still pulling in the caption. It's not 100% of the Post, but I think it's a representative sample. I just got a quick education in Markdown. ;)

Thanks! Dave

`[caption id="attachment_422" align="alignright" width="300" caption="Doctor Delts didn&#39;t want me to use his picture. He says he&#39;s not ripped enough yet, but he looks very similar to this fellow."]<img class="size-full wp-image-422" title="Doctor Delts" src="http://fitnesstabby.com/wp-content/uploads/2011/08/MuscleGuyDumbbells.jpg" alt="" width="300" height="375" />[/caption]`

`I'm lucky to be very self-disciplined, and have trained on my own for years.  But many people enjoy having workout partners, and now I occasionally have company on my workouts.  Doctor Delts, my favorite workout partner, is not your typical doctor, but more of a doctor of distractibility, and a kind of physician of flirtatiousness.  We get along great even though his personality and workout style are quite different from mine.  After having gone through a stage of doing a whole lot of pressing exercises, he has developed the shoulders that gave him his nickname.`
`<h2>Four Gyms in One Day</h2>`
`Delts is extremely enthusiastic,  but not overly patient.  Just last Wednesday, I met up with him at our favorite park for an outdoor workout.  He had gotten there before me, and was beaming with pride as he held a small woman in the air.  Though she looked a little concerned, she seemed to be taking being mildly airborne surprisingly well, and even smiling a little.  He does have a certain mischievous charm.  After he set her down and flirted with her a bit, she left.  He excitedly told me that he'd joined four gyms.  He is not independently wealthy, and I asked him how he was planning to pay for it.  He paused and frowned, and then shrugged, "I'm not... really sure about that.  But I'm thinking I can just try 'em, and I'll quit the ones I don't like, right?  They all sounded pretty good".`
